sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,211 @@ def transform(self, X):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : array or sparse matrix, shape = [n_samples, n_features]
+             The missing indicator for input data
+
+        """
+        self._validate_params()
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64, force_all_finite=False)
+
+        imputer_mask, feat_with_missing = self._get_missing_features_info(X)
+
+        self.n_features_ = X.shape[1]
+        if isinstance(self.features, six.string_types):
+            if self.features == "train":
+                self.feat_with_missing_ = feat_with_missing
+            else : # self.features == "all"
+                self.feat_with_missing_ = np.arange(self.n_features_)
+        else :
+            self.feat_with_missing_ = self.features
+
+        if not (isinstance(self.features, six.string_types) and self.features == "all"):
+            # no need to slice when all features have missing values
+            imputer_mask = imputer_mask[:, self.feat_with_missing_]
+
+        return imputer_mask
+
+    def _validate_params(self):
+        if (isinstance(self.features, six.string_types) and
+                self.features not in ["train", "all"]):
+            raise ValueError("Can only use these options: 'train', 'all'"
+                             " got {0}".format(self.features))
+        elif not isinstance(self.features, six.string_types):
+            self.features = check_array(self.features, ensure_2d=False)
+            if (isinstance(self.features, np.ndarray) and
+                    self.features.dtype.kind != 'i'):
+                raise ValueError("Features should be an array of integers")
+
+        if not ((isinstance(self.sparse, six.string_types) and
+                self.sparse == "auto") or isinstance(self.sparse, bool)):
+            raise ValueError("sparse can only boolean or 'auto'"
+                             " got {0}".format(self.sparse))
+
+    def _get_missing_features_info(self, X):
+        if sparse.issparse(X) and self.missing_values != 0:
+            #  sparse matrix and missing values is not zero
+            imputer_mask = _get_mask(X.data, self.missing_values)
+            imputer_mask = X.__class__((imputer_mask, X.indices.copy(),
+                                        X.indptr.copy()), shape=X.shape,
+                                       dtype=X.dtype)
+            feat_with_missing = imputer_mask.sum(axis=0).nonzero()[1]
+        else:
+            #  sparse with zero as missing value and dense matrix
+            if sparse.issparse(X):
+                X = X.toarray()
+            imputer_mask = _get_mask(X, self.missing_values)
+            #  convert boolean mask to binary mask
+            imputer_mask = astype(imputer_mask, int, copy=False)
```

Comment:
astype no longer exists in fixes due to the increase in numpy dependency. Use imputer_mask.astype(int, copy=False)